protostar/starknet_gateway/multicall/call_resolver.py

```diff
@@ -0,0 +1,82 @@
+from typing import Union
+
+from starknet_py.net.udc_deployer.deployer import Deployer
+
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet import Address
+from protostar.starknet.selector import Selector
+
+from .multicall_structs import (
+    CallBase,
+    InvokeCall,
+    DeployCall,
+    DeployCallName,
+    MulticallInputCalldata,
+    ResolvedCall,
+)
+
+
+class CallResolver:
```

Comment:
The resolving idea seems nice, but could we make it more generic? So that a user could define some `$variable` in a file and i.e. pass it as a parameter to the script? We could raise on a missing parameter value. Of course this can be in a separate PR but i'd think a bit about the syntax here - a name on itself might be confusing to read. People are familiar with this kind of variable interpolation, so perhaps a deploy could have a "bind_to": "$address1" instead of "name" and then the concept is becoming more universal than just addresses. WDYT?